core/cfs/cfs-coffee.c

```diff
@@ -1063,7 +1063,12 @@ cfs_seek(int fd, cfs_offset_t offset, int whence)
   }
 
   if(fdp->file->end < new_offset) {
-    fdp->file->end = new_offset;
+    if (FD_WRITABLE(fd)) {
+      fdp->file->end = new_offset;
+    } else {
+      /* Disallow seeking past the end of the file for read only FDs */
+      new_offset = fdp->file->end;
```

Comment:
I prefer to <code>return (cfs_offset_t)-1</code> when the seeking fails so it is clear to the caller.